### DISCUSSION TITLE
noobs guide

### DISCUSSION THREAD

**Discussion by dan-bar-dov** [emacs-lsp/lsp-mode#2968] [Q&A] [answered]
I am new to lsp, and find the documentation lacking for how to get started.

I have a small project, single directory, 3 C files, 1 H file. lsp-session-describe shows them all.
Yet, when I am trying to xref-find-definition of a function that is in another file, I get "no definition found".

Some explanation about what is a "project", "session", how to manage compilation flags (include paths), a "getting started" that is about how to use, (not how to configure the various options) would be nice.

For now, if someone can tip me on how such a small project can be cross referenced, would be great.

**Comment by yyoncho** [comment, marked as answer]
Have you checked https://emacs-lsp.github.io/lsp-mode/tutorials/CPP-guide/  ?

I am not 100% sure if clangd can handle simple projects without compile_commands.json - @sebastiansturm @danielmartin, do you have a clue?

> Some explanation about what is a "project", "session", how to manage compilation flags 

I have this on my TODO list.

**Reply by dan-bar-dov** [reply]
Thanks for replying. I went again over the CPP-guide, made sure I have the .emacs setup properly, and restarted emacs.
Works the same. No cross references between files.

This comment about clangd and compile_commands.json did the trick. After I created this file (using bear), cross references work. 

Thanks
